scripts/postbuild.ts

```diff
@@ -0,0 +1,61 @@
+/*
+  This script prepares the central `build` directory for NPM package creation.
+  It first copies all non-code files into the `build` directory, including `package.json`, which
+  is edited to adjust entry point paths. These corrections are performed so that the paths align with
+  the directory structure inside `build`.
+*/
+
+import * as fs from 'fs';
+
+import * as path from 'path';
+
+const BUILD_DIR = 'build';
+const ASSETS = ['README.md', 'LICENSE', 'package.json', '.npmignore'];
+const ENTRY_POINTS = ['main', 'module', 'types'];
+
+// check if build dir exists
+try {
+  if (!fs.existsSync(path.resolve(BUILD_DIR))) {
+    console.error(`Directory ${BUILD_DIR} DOES NOT exist`);
+    console.error("This script should only be executed after you've run `yarn build`.");
+    process.exit(1);
+  }
+} catch (error) {
+  console.error(`Error while looking up directory ${BUILD_DIR}`);
+}
+
+// copy non-code assets to build dir
+ASSETS.forEach(asset => {
+  const assetPath = path.resolve(asset);
+  try {
+    if (!fs.existsSync(assetPath)) {
+      console.error(`Asset ${asset} does not exist.`);
+      process.exit(1);
+    }
+    fs.copyFileSync(assetPath, path.resolve(BUILD_DIR, asset));
+  } catch (error) {
+    console.error(`Error while copying ${asset} to ${BUILD_DIR}`);
+  }
+});
+
+// package.json modifications
+const packageJsonPath = path.resolve(BUILD_DIR, 'package.json');
+const pkgJson: { [key: string]: string } = require(packageJsonPath);
+
+// modify entry points to point to correct paths (i.e. strip out the build directory)
+ENTRY_POINTS.filter(entryPoint => pkgJson[entryPoint]).forEach(entryPoint => {
+  pkgJson[entryPoint] = pkgJson[entryPoint].replace(`${BUILD_DIR}/`, '');
```

Comment:
nit: filter + ForEach can just be a single reduce call, also means we emit a new object instead of mutating the previous reference.

Reply:
> also means we emit a new object instead of mutating the previous reference.

`.filter()` already creates a new array. 

I also think that doing it in two stages like this makes it easier to reason about compared to using `.reduce()`.  Of course, we could also always just do `ENTRY_POINTS.forEach(entryPoint => if pkgJson[entryPoint] { <make value change> }})`. Up to you, @Lms24.

Reply:
Thanks for the feedback but  I don't entirely understand the reasoning of changing this. I used the `filter`-then-`foreach` approach to make it clear what my intention is here (i.e. only change entry point paths on properties that in fact exist in the `package.json` file). Which IMHO (love to be proven wrong though) is a quite clean approach of operating on arrays. Also, I don't entirely see how reduce would be nicer here, could you explain?
Unless there's a good reason which I overlooked (which there very well might be), I'd leave it as is (also keeping in mind our moving fast approach...)

Reply:
Something like so (please double check I wrote this in GH)

```js
const newPkgJson = ENTRY_POINTS.reduce((acc, curr) => {
  acc[curr] = acc[curr].replace(`${BUILD_DIR}/`, '');
  return acc;
}, pkgJson)
```

Reply:
> Something like so (please double check I wrote this in GH)
> 
> ```js
> const newPkgJson = ENTRY_POINTS.reduce((acc, curr) => {
>   acc[curr] = acc[curr].replace(`${BUILD_DIR}/`, '');
>   return acc;
> }, pkgJson)
> ```

I find this _significantly_ harder to understand as a reader, both in and of itself and as a way to communicate the intention here.

Reply:
Ah interesting - I find it much easier to understand, it's just an array reducing to an object (a very typical FP operation) - I'm fine with leaving as is though.

Reply:
Go with your original, @Lms24. You have the power!

Reply:
> it's just an array reducing to an object (a very typical FP operation)

@AbhiPrasad - perhaps that explains it. Because I haven't done a ton of FP, it doesn't hit my brain as, "Oh, that matches a pattern I know." Instead, my internal monologue is more like, "Okay, we're reducing, so we're combining stuff. How are we combining it? Oh, we're not, we're just fixing a string, but wait, is the string already there? Are we starting with something non-empty? `acc` kinda implies starting empty... Let's see - ah, okay, no, it's not empty. We're starting the the `package.json` data. Okay, but so wait, are we changing every entry? Where is `curr` coming from? The name doesn't tell me, so lemme look backwards... okay, yes, right - we're not iterating over `package.json` entries, we're iterating over `ENTRY_POINTS`. So `curr` is an entry point... ohhhh, okay, right. I get it. We're changing the value for all of the entry points."

vs

"Okay, we're taking the entry points, oh, but only the ones which exist in package.json, and then we're changing their value. Okay, got it."